### PR TITLE
Adding collection.ConfigSest specification to collection CRD

### PR DESCRIPTION
**Pull request by sepulworld**
Signed-off-by: Zane Williamson <zanew@zillow.com>

**Describe your changes**
Adding a required spec entry for the Collection CRD:  `collectionConfigName`

The only Collection API endpoint that contained the status of a collection's configset was found in  [CLUSTERSTATUS](https://lucene.apache.org/solr/guide/8_1/collections-api.html#clusterstatus)

Instead of making a call to both COLSTATUS and CLUSTERSTATUS, we can just make one call to CLUSTERSTATUS (specifying the collection) and get everything we need. So in  this PR moved over to using CLUSTERSTATUS instead of COLSTATUS.

Also, updated the README for some notes on setting up Docker for Mac with K8s enabled.

**Testing performed**
Local and development environment testing done with examples all work as expected.

**Additional context**
I will open a PR to have us start a discussion on possible implementations of a ConfigSet CRD. Where we could have the operator manage the lifecycle of config sets for us.

Until then, this PR will allow us to create a collection using `collectionConfigName: "_default"` in the Spec. Then we can use the SolrCloud API outside of the solr-operator to create a new configset which we can switch to after it is in place by updating the collectionConfigName spec entry.

### CODE REVIEW

**Comment by sepulworld** [comment]
Odd, build failed with:

```
package sigs.k8s.io/kustomize/cmd/kustomize: cannot find package "sigs.k8s.io/kustomize/cmd/kustomize" in any of:
	/home/travis/.gimme/versions/go1.12.linux.amd64/src/sigs.k8s.io/kustomize/cmd/kustomize (from $GOROOT)
	/home/travis/gopath/src/sigs.k8s.io/kustomize/cmd/kustomize (from $GOPATH)
The command "./hack/install_dependencies.sh" exited with 1.
```

I don't see this locally when I run `make docker-base-build docker-build docker-push` Will look closer, but not related to code changes in the PR.

**Comment by sepulworld** [comment]
Appears to be happening on the TravisCI container at this step before Make driven docker steps

https://github.com/bloomberg/solr-operator/blob/master/.travis.yml#L12

**Comment by sepulworld** [comment]
Ok, looks like Kustomize moved over to supporting Go modules and in the process updated the path for go getting. 

https://github.com/kubernetes-sigs/kustomize/commit/98d1893057b35efc3aae759bfd525c6a21419a8b

**Comment by sepulworld** [comment]
Ok, fixed the issue with Kustomize installation blocking build.

File: README.md [hunk 1]
```diff
@@ -309,6 +312,30 @@ Building and releasing a test operator image with a custom namespace.
 $ NAMESPACE=your-namespace make docker-base-build docker-build docker-push
 ```
 
+### Docker for Mac Local Development Setup
```
Comment (HoustonPutman) [inline README.md]: Awesome section to add

File: pkg/apis/solr/v1beta1/solrcollection_types.go [hunk 2]
```diff
@@ -28,6 +28,9 @@ type SolrCollectionSpec struct {
 	// The name of the collection to perform the action on
 	Collection string `json:"collection"`
 
+	// Define a configset to use for the collection. If not provided _default will be used and an AUTOCREATED mutable config will be generated
+	CollectionConfigName string `json:"collectionConfigName"`
```
[the remark this answers is not in the document]
Comment (sepulworld) [reply pkg/apis/solr/v1beta1/solrcollection_types.go]: Actually, I was intending to make this a required parameter for a collection. I updated the comments to indicate as such. Hope that works for you, or I can look into making this optional.

[the remark this answers is not in the document]
Comment (HoustonPutman) [reply pkg/apis/solr/v1beta1/solrcollection_types.go]: Nope thats fine, and we can always make it optional later if we want to.

**Pull request merged**